edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java

```diff
@@ -217,20 +217,23 @@ public void setCalculatedPointsEarned(final double literalTotalPointsEarned) {
 	}
 
 	public void initNonpersistentFields(final double totalPointsPossible, final double totalPointsEarned,
-			final double literalTotalPointsEarned) {
-		Double percentageEarned;
+			final double literalTotalPointsEarned, final double totalPointsExtra) {
+		BigDecimal percentageEarned;
+		BigDecimal percentageExtraEarned;
 		this.calculatedPointsEarned = literalTotalPointsEarned;
 		this.totalPointsPossible = totalPointsPossible;
 		final BigDecimal bdTotalPointsPossible = new BigDecimal(totalPointsPossible);
 		final BigDecimal bdTotalPointsEarned = new BigDecimal(totalPointsEarned);
 
 		if (totalPointsPossible <= 0.0) {
-			percentageEarned = null;
+			this.autoCalculatedGrade = null;
 		} else {
-			percentageEarned = Double.valueOf(bdTotalPointsEarned.divide(bdTotalPointsPossible, GradebookService.MATH_CONTEXT)
-					.multiply(new BigDecimal("100")).doubleValue());
+			percentageEarned =  bdTotalPointsEarned.divide(bdTotalPointsPossible, GradebookService.MATH_CONTEXT)
```

Comment:
Would I use MathContext.DECIMAL128? like in this one
https://github.com/sakaiproject/sakai/commit/a3a54b1e408e73a06550e67e77552a1f55e1974f#diff-290359684d1bbbe62fe25dcaeb0f85fdL1302

Reply:
No, I don't see any need to use MathContext.DECIMAL128